src/mono/wasm/runtime/driver.c

```diff
@@ -111,13 +113,14 @@ mono_wasm_invoke_js (MonoString *str, int *is_exception)
 		}
 		var buff = Module._malloc((res.length + 1) * 2);
 		stringToUTF16 (res, buff, (res.length + 1) * 2);
+		setValue ($3, res.length, "i32");
 		return buff;
-	}, (int)native_val, native_len, is_exception);
+	}, (int)native_val, native_len, is_exception, p_native_res_len);
```

Comment:
ah, I was wondering how to solve this problem without multiple return values. good solution.